### DISCUSSION TITLE
MQTT Topic based on FriendlyName

### DISCUSSION THREAD

**Discussion by bugsyb** [arendst/Tasmota#21507] [Q&A] [answered]
Noob to Tasmota here and need some help. Did RTFM and code check.

The default MQTT Topic set to `tasmota_%06X` needs to be changed for me to something like `tasmota_%DeviceName%` or FriendlyName. Tried multiple variations of the variable and can't nail, how to use variable instead of a static string.

The target is to have all plugs named, i.e. "tasmota_fridge", "tasmota_something", based on Friendly Name. The plug mac address/id might be changing over time as plugs might be changed and as data is pumped to InfluxDB and use then later, naming needs to be consistent for various reasons.

Thanks for help!

**Comment by sfromis** [comment, marked as answer]
Standard setup procedures does not support basing names on other names. And I fully agree with not using the default scheme which just has the purpose of avoiding the mess from having multiple devices with same names clashing.

Other than obviously just configuring the names you want, per device, you could create some sort of automation detecting names in need of change, and generating commands for that.

**Comment by bugsyb** [comment]
Thank you.
